test/units/Node-events.ts

```diff
@@ -0,0 +1,56 @@
+import { DOMParser } from "../../deno-dom-wasm.ts";
+import {
+  assert,
+  assertEquals,
+} from "https://deno.land/std@0.85.0/testing/asserts.ts";
+
+Deno.test("Node.dispatchEvent", () => {
+  const doc = new DOMParser().parseFromString(
+    `<div></div>`,
+    "text/html",
+  )!;
+  const div = doc.querySelector("div")!;
+  let called = false;
+  div.addEventListener("click", () => {
+    called = true;
+  });
+  div.dispatchEvent(new Event("click"));
+  assert(called);
+});
+
+Deno.test("Node.dispatchEvent, event bubbles", async () => {
+  const doc = new DOMParser().parseFromString(
+    `<div class="parent"><div class="child"></div></div>`,
+    "text/html",
+  )!;
+  const parent = doc.querySelector(".parent")!;
+  const child = doc.querySelector(".child")!;
+  let called = false;
+  // FIXME(kt3k): this empty event handler is necessary to work around
+  // the bug in EventTarget impl in Deno.
```

Comment:
Is there an issue open for this? Maybe we can include it in the comment here

Reply:
No issue yet, but this line causes the problem https://github.com/denoland/deno/blob/07618c861e27e284cae3a309783ef78dcef611dd/ext/web/02_event.js#L976-L979

It only checks the dispatched element's handler types no matter its ancestors have handlers of the given type